tests/unit/test_signature/test_parsing.py

```diff
@@ -284,3 +289,229 @@ def handler(param: bool) -> None:
+        int_header: int = Parameter(header="X-SOME-INT"),
+        int_cookie: int = Parameter(cookie="int-cookie"),
+    ) -> None:
+        ...
+
+    with create_test_client(route_handlers=[test]) as client:
+        response = client.post(
+            "/",
+            json={"child": {"val": "a", "other_val": "b"}, "other_child": {"val": [1, "c"]}},
+            params={"int_param": "param", "length_param": "d"},
+            headers={"X-SOME-INT": "header"},
+            cookies={"int-cookie": "cookie"},
+        )
+
+        assert response.status_code == HTTP_400_BAD_REQUEST
+
+        data = response.json()
+
+        assert data
+        assert data["extra"] == [
+            {"key": "child.val", "message": "value is not a valid integer", "source": "body"},
+            {"key": "child.other_val", "message": "value is not a valid integer", "source": "body"},
+            {"key": "other_child.val.1", "message": "value is not a valid integer", "source": "body"},
+            {"key": "int_param", "message": "value is not a valid integer", "source": "query"},
+            {"key": "length_param", "message": "ensure this value has at least 2 characters", "source": "query"},
+            {"key": "int_header", "message": "value is not a valid integer", "source": "header"},
+            {"key": "int_cookie", "message": "value is not a valid integer", "source": "cookie"},
+        ]
+
+
+def test_invalid_input_attrs() -> None:
+    @define
+    class OtherChild:
+        val: List[int]
+
+    @define
+    class Child:
+        val: int
+        other_val: int
+
+    @define
+    class Parent:
+        child: Child
+        other_child: OtherChild
+
+    @post("/")
+    def test(
+        data: Parent,
+        int_param: int,
+        int_header: int = Parameter(header="X-SOME-INT"),
+        int_cookie: int = Parameter(cookie="int-cookie"),
+    ) -> None:
+        ...
+
+    with create_test_client(route_handlers=[test]) as client:
+        response = client.post(
+            "/",
+            json={"child": {"val": "a", "other_val": "b"}, "other_child": {"val": [1, "c"]}},
+            params={"int_param": "param"},
+            headers={"X-SOME-INT": "header"},
+            cookies={"int-cookie": "cookie"},
+        )
+
+        assert response.status_code == HTTP_400_BAD_REQUEST
+
+        data = response.json()
+
+        assert data
+        assert data["extra"] == [
+            {"key": "child.val", "message": "invalid literal for int() with base 10: 'a'", "source": "body"},
+            {"key": "child.other_val", "message": "invalid literal for int() with base 10: 'b'", "source": "body"},
+            {"key": "other_child.val.1", "message": "invalid literal for int() with base 10: 'c'", "source": "body"},
+            {"key": "int_param", "message": "invalid literal for int() with base 10: 'param'", "source": "query"},
+            {"key": "int_header", "message": "invalid literal for int() with base 10: 'header'", "source": "header"},
```

Comment:
Note that there's no `length_param` `extra` value here like there are for the other validation types. I'm noticing that when you have a param with no default followed by a param with a default and the no-default one fails validation, the one with a default won't validate fully (e.g., the min_length won't be validated against, but the type will).

This can be validated by changing `tests/unit/test_params.py::test_parsing_of_parameter_as_default_value` to the following and viewing the output of `response.json()`:

```python
@pytest.mark.parametrize("backend", ("pydantic", "attrs"))
def test_parsing_of_parameter_as_default_value(backend: Any) -> None:
    @get(path="/")
    def handler(int_param: int, other_int_param: int, param: int = Parameter(ge=1)) -> str:
        return param

    with create_test_client(handler, _preferred_validation_backend=backend) as client:
        response = client.get("/?int_param=a&other_int_param=a&param=a")
        assert response.status_code == HTTP_400_BAD_REQUEST
        print(f'{response.json()=}')

        response = client.get("/?int_param=4&other_int_param=4&param=a")
        assert response.status_code == HTTP_200_OK
```

This fails the `int` check. But if you change it to `param: str = Parameter(min_length=1)`, `param` will not fail validation.

Reply:
Sure, make a ticket?

Reply:
Will do!